crates/uv/tests/it/python_find.rs

```diff
@@ -707,3 +707,190 @@ fn python_required_python_major_minor() {
     error: No interpreter found for Python >3.11.[X], <3.12 in virtual environments, managed installations, or search path
     "###);
 }
+
+#[test]
+fn python_find_script() {
+    let context = TestContext::new("3.13");
+    let filters = context
+        .filters()
+        .into_iter()
+        .chain(vec![(
+            r"environments-v2/[\w-]+",
+            "environments-v2/[HASHEDNAME]",
+        )])
+        .collect::<Vec<_>>();
+
+    uv_snapshot!(filters, context.init().arg("--script").arg("foo.py"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    Initialized script at `foo.py`
+    "###);
+
+    uv_snapshot!(filters, context.sync().arg("--script").arg("foo.py"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    Creating script environment at: [CACHE_DIR]/environments-v2/[HASHEDNAME]
+    Resolved in [TIME]
+    Audited in [TIME]
+    "###);
+
+    if cfg!(windows) {
+        uv_snapshot!(filters, context.python_find().arg("--script").arg("foo.py"), @r###"
+        success: true
+        exit_code: 0
+        ----- stdout -----
+        [CACHE_DIR]/environments-v2/[HASHEDNAME]/Scripts/python.exe
+
+        ----- stderr -----
+        "###);
+    } else {
+        uv_snapshot!(filters, context.python_find().arg("--script").arg("foo.py"), @r###"
+        success: true
+        exit_code: 0
+        ----- stdout -----
+        [CACHE_DIR]/environments-v2/[HASHEDNAME]/bin/python3
+
+        ----- stderr -----
+        "###);
+    }
+}
+
+#[test]
+fn python_find_script_no_environment() {
+    let context = TestContext::new("3.13");
+
+    let script = context.temp_dir.child("foo.py");
+
+    script
+        .write_str(indoc! {r"
+            # /// script
+            # dependencies = []
+            # ///
+        "})
+        .unwrap();
+
+    if cfg!(windows) {
+        uv_snapshot!(context.filters(), context.python_find().arg("--script").arg("foo.py"), @r###"
+        success: true
+        exit_code: 0
+        ----- stdout -----
+        [VENV]/Scripts/python.exe
```

Comment:
Oh it's actually quite surprising we'd pull the Python from the virtual environment here. I wonder if we should avoid selecting active virtual environments during script Python discovery? That's a separate issue from this PR though.